Diagnostic/diagnostic.py

```diff
@@ -648,14 +649,27 @@ def start_mdsd():
             last_error_time = datetime.datetime.now()
             while True:
                 time.sleep(30)
-                if " ".join(get_mdsd_process()).find(str(mdsd.pid)) <0 and len(get_mdsd_process()) >=2:
+                if " ".join(get_mdsd_process()).find(str(mdsd.pid)) < 0 and len(get_mdsd_process()) >= 2:
                     mdsd.kill()
                     hutil.log("Another process is started, now exit")
                     return
-                if not (mdsd.poll() is None):
+                if mdsd.poll() is not None:     # if mdsd has terminated
                     time.sleep(60)
                     mdsd_log.flush()
                     break
+
+                # mdsd is now up for at least 30 seconds.
+
+                # Issue #128 LAD should restart OMI if it crashes
+                omi_up_and_running = RunGetOutput("/opt/omi/bin/omicli noop")[0] is 0
+                if not omi_up_and_running:
+                    hutil.error("OMI noop query failed. OMI crash suspected. Restarting OMI and send SIGHUP to mdsd after 5 seconds.")
+                    omi_restart_msg = RunGetOutput("/opt/omi/bin/service_control restart")[1]
+                    hutil.error("OMI restart result: " + omi_restart_msg)
```

Comment:
Makes sense. Will add the code to check that as the condition of mdsd.send_signal(signal.SIGHUP). Thx.